src/feature_extractor/hr_to_csv.py

```diff
@@ -0,0 +1,215 @@
+        help="Method for ROI extraction.",
+        choices=["skin_adapt", "skin_fix", "rect"])
+
+    parser.add_argument(
+        "-sta", "--skin-thresh-adapt",
+        type=float,
+        help="Skin threshold needed when \
+        type-roi=skin_adapt.")
+
+    parser.add_argument(
+        "-stf", "--skin-thresh-fix",
+        type=float,
+        help="Skin threshold needed when \
+        type-roi=skin_fix.")
+
+    parser.add_argument(
+        "-rc", "--rect_coords",
+        type=list,
+        help="ROI coordinates, needed when \
+            type-roi=skin_fix and no \
+            rect-regions is provided.")
+
+    parser.add_argument(
+        "-rr", "--rect-regions",
+        type=list,
+        help="ROI regions, needed when type-roi=rect \
+            and no rect-coords is provided.",
+        choices=["forehead", "lcheek", "rcheek", "nose"])
+
+    return parser.parse_args()
+
+
+def ica_estimator(args):
+    face_extractor = VideoWrapper(
+        video_path=args.video_path,
+        detector=args.detector,
+        extractor=args.extractor,
+        type_roi=args.type_roi,
+        skin_thresh_adapt=args.skin_thresh_adapt,
+        skin_thresh_fix=args.skin_thresh_fix,
+        rect_coords=args.rect_coords,
+        rect_regions=args.rect_regions)
+
+    face_extractor.extract_faces(verbose=True)
+    face_extractor.show_faces()
+    ica = ICAEstimator(face_extractor)
+    ica.run_offline()
+
+    return ica.bpm_estimation
+
+
+def append_heart_rate(output_file, bpm_estimations):
+    dataset = pd.read_csv(
+        output_file, float_precision="round_trip",
+        usecols=range(0, AMOUNT_OF_FEATURES))
+
+    dataset_rows_amount = dataset.shape[0]
+    dataset_columns_amount = dataset.shape[1]
+    hr_estimation_amount = len(bpm_estimations)
+
+    if dataset_rows_amount != hr_estimation_amount:
+        raise Exception(
+            "Dataset samples amount and heart rate " +
+            "estimations must match in lenght")
+
+    dataset.insert(dataset_columns_amount, "hr", bpm_estimations)
+
+    dataset.to_csv(output_file, float_format="%.4f", index=False)
+
+
+def show_estimator_config(args):
+    d = args.__dict__
+    max_str_len = max([len(val) for val in d.values() if isinstance(val, str)])
+    print("=========================={}".format("="*max_str_len))
+    print(" File name            : ", args.video_path)
+    print(" Output CSV file      : ", args.output_path)
+    print(" Detector             : ", args.detector)
+    print(" Extractor            : ", args.extractor)
+    print(" Type ROI             : ", args.type_roi)
+    print(" Skin threshold adapt : ", args.skin_thresh_adapt)
+    print(" Skin threshold fix   : ", args.skin_thresh_fix)
+    print(" Rectangle coordinates: ", args.rect_coords)
+    print(" Rectangle regions    : ", args.rect_regions)
+    print("=========================={}".format("="*max_str_len))
+
+
+def dummy_test(args):
+    """
+    To run this test a csv with features
```

Comment:
This method it's intended for testing, when everything is 100% we must remove him